tool/grind.dart

```diff
@@ -138,49 +126,33 @@ Directory get testPackageExperiments =>
 Directory get pluginPackage => Directory(path
     .joinAll(['testing', 'flutter_packages', 'test_package_flutter_plugin']));
 
-Directory _testPackageDocsDir;
-
-Directory get testPackageDocsDir =>
-    _testPackageDocsDir ??= createTempSync('test_package');
-
-Directory _testPackageExperimentsDocsDir;
-
-Directory get testPackageExperimentsDocsDir =>
-    _testPackageExperimentsDocsDir ??=
-        createTempSync('test_package_experiments');
+late final Directory _testPackageDocsDir = createTempSync('test_package');
 
-Directory _pluginPackageDocsDir;
+late final Directory _testPackageExperimentsDocsDir =
+    createTempSync('test_package_experiments');
 
-Directory get pluginPackageDocsDir =>
-    _pluginPackageDocsDir ??= createTempSync('test_package_flutter_plugin');
+late final String _pluginPackageDocsPath =
+    createTempSync('test_package_flutter_plugin').path;
 
 /// Version of dartdoc we should use when making comparisons.
 String get dartdocOriginalBranch {
   var branch = 'master';
   if (Platform.environment.containsKey('DARTDOC_ORIGINAL')) {
```

Comment:
We could extract `Platform.environment['DARTDOC_ORIGINAL']` as a local variable, and check it for null instead of using `containsKey`.